datasource.go

```diff
@@ -98,11 +99,20 @@ func (ds *sqldatasource) handleQuery(req backend.DataQuery) (data.Frames, error)
 		fillMode = q.FillMissing
 	}
 
+	// The database connection may vary depending on query arguments
+	db := ds.db
+	if ds.DB != nil {
+		db, err = ds.DB(q)
```

Comment:
We will probably need to close this database when the request is complete, right?

Just a simple `defer db.Close()` here might suffice.

Reply:
actually, it may be nice to reuse these db instances. Let me re-work this to support this (and do a clean-up in the `Dispose` method).